pkg/odo/genericclioptions/resolve.go

```diff
@@ -91,11 +88,7 @@ func resolveComponent(cmdline cmdline.Cmdline, localConfiguration localConfigPro
 }
 
 func resolveProject(cmdline cmdline.Cmdline, localConfiguration localConfigProvider.LocalConfigProvider) string {
-	projectFlag := cmdline.FlagValueIfSet(util.ProjectFlagName)
-	if projectFlag != "" {
-		return projectFlag
-	}
-	return localConfiguration.GetNamespace()
+	return cmdline.FlagValueIfSet(util.ProjectFlagName)
```

Comment:
You might as well just call Cmdline.FlagValueIfSet method where you're calling resolveProject. If not, you can consider removing localConfiguration arg from the function signature.

Reply:
I removed the unnecessary param. I prefer to keep the function, to hide the details to the caller